### PR TITLE
fix(components): [time-picker] click the Cancel button twice to close time picker

### LINKED ISSUE #18463: [Component] [time-picker] el-time-picker设置is-range后，选择时间后，点击取消按钮，会恢复上一次的时间或默认时间，然后需要再点一次才能关闭选择器弹窗
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.21`
- Element Plus Version: `2.8.4`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-time-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZGVtby1yYW5nZVwiPlxuICAgIDxlbC10aW1lLXBpY2tlclxuICAgICAgdi1tb2RlbD1cInZhbHVlMVwiXG4gICAgICBpcy1yYW5nZVxuICAgICAgcmFuZ2Utc2VwYXJhdG9yPVwiVG9cIlxuICAgICAgc3RhcnQtcGxhY2Vob2xkZXI9XCJTdGFydCB0aW1lXCJcbiAgICAgIGVuZC1wbGFjZWhvbGRlcj1cIkVuZCB0aW1lXCJcbiAgICAvPlxuICAgIDxlbC10aW1lLXBpY2tlclxuICAgICAgdi1tb2RlbD1cInZhbHVlMlwiXG4gICAgICBpcy1yYW5nZVxuICAgICAgYXJyb3ctY29udHJvbFxuICAgICAgcmFuZ2Utc2VwYXJhdG9yPVwiVG9cIlxuICAgICAgc3RhcnQtcGxhY2Vob2xkZXI9XCJTdGFydCB0aW1lXCJcbiAgICAgIGVuZC1wbGFjZWhvbGRlcj1cIkVuZCB0aW1lXCJcbiAgICAvPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IHZhbHVlMSA9IHJlZjxbRGF0ZSwgRGF0ZV0+KFtcbiAgbmV3IERhdGUoMjAxNiwgOSwgMTAsIDgsIDQwKSxcbiAgbmV3IERhdGUoMjAxNiwgOSwgMTAsIDksIDQwKSxcbl0pXG5jb25zdCB2YWx1ZTIgPSByZWY8W0RhdGUsIERhdGVdPihbXG4gIG5ldyBEYXRlKDIwMTYsIDksIDEwLCA4LCA0MCksXG4gIG5ldyBEYXRlKDIwMTYsIDksIDEwLCA5LCA0MCksXG5dKVxuPC9zY3JpcHQ+XG5cbjxzdHlsZT5cbi5kZW1vLXJhbmdlIC5lbC1kYXRlLWVkaXRvciB7XG4gIG1hcmdpbjogOHB4O1xufVxuXG4uZGVtby1yYW5nZSAuZWwtcmFuZ2Utc2VwYXJhdG9yIHtcbiAgYm94LXNpemluZzogY29udGVudC1ib3g7XG59XG48L3N0eWxlPlxuIiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguY3NzJywgJ2h0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC90aGVtZS1jaGFsay9kYXJrL2Nzcy12YXJzLmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

1. 打开弹窗选择器
2. 选择跟弹窗初始默认时间不一样的时间
3. 点击取消按钮
4. 会发现弹窗选择器没有关闭，时间变回弹窗初始默认时间
5. 再点取消按钮，才关闭弹窗选择器

### What is Expected?

第一次点取消按钮的时候关闭弹窗选择器

### What is actually happening?

需要点两次

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by thinkasany**
关联pr： #18908  #18823

closed https://github.com/element-plus/element-plus/issues/18463

### CODE REVIEW

**Comment by thinkasany** [comment]
[运行环境](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZGVtby1yYW5nZVwiPlxuICAgIDxlbC10aW1lLXBpY2tlclxuICAgICAgdi1tb2RlbD1cInZhbHVlMVwiXG4gICAgICBpcy1yYW5nZVxuICAgICAgcmFuZ2Utc2VwYXJhdG9yPVwiVG9cIlxuICAgICAgc3RhcnQtcGxhY2Vob2xkZXI9XCJTdGFydCB0aW1lXCJcbiAgICAgIGVuZC1wbGFjZWhvbGRlcj1cIkVuZCB0aW1lXCJcbiAgICAvPlxuICAgIDxlbC10aW1lLXBpY2tlclxuICAgICAgdi1tb2RlbD1cInZhbHVlMlwiXG4gICAgICBpcy1yYW5nZVxuICAgICAgYXJyb3ctY29udHJvbFxuICAgICAgcmFuZ2Utc2VwYXJhdG9yPVwiVG9cIlxuICAgICAgc3RhcnQtcGxhY2Vob2xkZXI9XCJTdGFydCB0aW1lXCJcbiAgICAgIGVuZC1wbGFjZWhvbGRlcj1cIkVuZCB0aW1lXCJcbiAgICAvPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IHZhbHVlMSA9IHJlZjxbRGF0ZSwgRGF0ZV0+KFtcbiAgbmV3IERhdGUoMjAxNiwgOSwgMTAsIDgsIDQwKSxcbiAgbmV3IERhdGUoMjAxNiwgOSwgMTAsIDksIDQwKSxcbl0pXG5jb25zdCB2YWx1ZTIgPSByZWY8W0RhdGUsIERhdGVdPihbXG4gIG5ldyBEYXRlKDIwMTYsIDksIDEwLCA4LCA0MCksXG4gIG5ldyBEYXRlKDIwMTYsIDksIDEwLCA5LCA0MCksXG5dKVxuPC9zY3JpcHQ+XG5cbjxzdHlsZT5cbi5kZW1vLXJhbmdlIC5lbC1kYXRlLWVkaXRvciB7XG4gIG1hcmdpbjogOHB4O1xufVxuXG4uZGVtby1yYW5nZSAuZWwtcmFuZ2Utc2VwYXJhdG9yIHtcbiAgYm94LXNpemluZzogY29udGVudC1ib3g7XG59XG48L3N0eWxlPlxuIiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9wcmV2aWV3LTE4OTM4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJywgJ2h0dHBzOi8vcHJldmlldy0xODkzOC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTg5MzgtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTE4OTM4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

**Comment by thinkasany** [comment]
@btea 佬，有空的时候帮忙看看这个问题。

**Comment by btea** [comment]
Could you add a test case?

**Comment by thinkasany** [comment]
> Could you add a test case?

这个单测好像不太好加，能否给点帮助。 我尝试通过复现两次关闭的mock去实现单测，就是先修改时间再去点击关闭，页面上的操作的话实际上是会先修改时间为 `oldValue`， 只有再次点击取消才会取消，但是单测中，一次就关闭了。我测试的时候移除了pr 中修改的代码。
```
  it('should select a time range and close the picker', async () => {
    const value = ref<[Date, Date]>([
      new Date(2016, 9, 10, 8, 40),
      new Date(2016, 9, 10, 9, 40),
    ])
    const wrapper = mount(() => (
      <TimePicker
        v-model={value.value}
        is-range
        start-placeholder="Start time"
        end-placeholder="End time"
      />
    ))
    const input = wrapper.find('input')
    input.trigger('blur')
    input.trigger('focus')
    await nextTick()
    // 修改左侧时间的小时
    const list = document.querySelectorAll('.el-time-spinner__list')
    const leftHoursEl = list[0]
    ;(leftHoursEl.querySelectorAll('.el-time-spinner__item')[7] as any).click() // 选择7小时
    await nextTick()

    // 检查时间选择器中显示的时间
    const selectedStartTime = value.value[0]
    expect(selectedStartTime.getHours()).toBe(7)

    // 点击取消按钮
    ;(document.querySelector('.el-time-panel__btn.cancel') as any).click()
    await rAF()

    // 检查时间选择器是否关闭
    expect(wrapper.find('.el-picker-panel').exists()).toBe(false)
    expect(document.querySelector('.el-time-spinner__item')).toBe(null)
    expect(document.querySelector('.el-time-panel__btn.cancel')).toBe(null)
  })
```

**Comment by btea** [comment]
或许可以验证一下点击取消按钮时是否触发滚动事件。

File: packages/components/time-picker/src/time-picker-com/panel-time-range.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div
-    v-if="actualVisible"
+    v-if="actualVisible || visible"
```
Comment (tolking) [inline packages/components/time-picker/src/time-picker-com/panel-time-range.vue:3]: Is there any need to modify this?

**Review by tolking — COMMENTED** [review]
For testing, you can [refer to here](https://github.com/element-plus/element-plus/blob/dev/packages/components/time-picker/__tests__/time-picker.test.tsx#L80-L133) to check whether the popup is closed.

File: packages/components/time-picker/src/time-picker-com/panel-time-range.vue (hunk 1 above)
Comment (thinkasany) [inline packages/components/time-picker/src/time-picker-com/panel-time-range.vue:3]:
```suggestion
    v-if="actualVisible"
```

**Review by tolking — APPROVED** [review]

**Review by btea — APPROVED** [review]
Thanks!

**Pull request merged**